crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -0,0 +1,187 @@
+//! This module defines Settlement as originated from a mined transaction
+//! calldata.
+
+use {
+    crate::{
+        boundary,
+        domain::{
+            auction::{self, order},
+            eth,
+        },
+    },
+    ethcontract::{common::FunctionExt, tokens::Tokenize, U256},
+    trade::Trade,
+};
+
+mod tokenized;
+mod trade;
+
+/// Settlement originated from a calldata of a settlement transaction.
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct Settlement {
+    trades: Vec<Trade>,
+    /// Data that was appended to the regular call data of the `settle()` call
+    /// as a form of on-chain meta data. This is used to associate a
+    /// settlement with an auction.
+    auction_id: auction::Id,
+}
+
+impl Settlement {
+    /// Number of bytes that may be appended to the calldata to store an auction
+    /// id.
+    pub const META_DATA_LEN: usize = 8;
```

Comment:
At least for now this doesn't need to be public.
```suggestion
    const META_DATA_LEN: usize = 8;
```

Reply:
fixed